### PR TITLE
[rust-server] (Re-)Adding support for rust-server

### PR DESCRIPTION
**Pull request by bjgill**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `3.1.x`, `4.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Description of the PR

This is an initial attempt at getting `rust-server` working in openapi-generator (and therefore fix https://github.com/OpenAPITools/openapi-generator/issues/111).

Thanks to @wing328 and @mirw for their help with this.

Outstanding issues:

- [ ] Responses with `schema: {type: string}`
- [ ] Enum strings (and arrays of strings) in form data params
- [ ] Byte strings in form data parameters
- [ ] Binary string properties

### CODE REVIEW

**Comment by bjgill** [comment]
@wing328 - opening as you suggested. The first issue above is the one currently causing the most problems.

File: samples/server/petstore/rust-server/src/models.rs [hunk 1]
```diff
@@ -139,38 +139,66 @@ impl ApiResponse {
...
 }
 
-#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
-pub struct ArrayOfArrayOfNumberOnly {
-    #[serde(rename = "ArrayArrayNumber")]
-    #[serde(skip_serializing_if="Option::is_none")]
-    pub array_array_number: Option<Vec<Vec<f64>>>,
+#[derive(Debug, Clone, PartialEq, PartialOrd, Serialize, Deserialize)]
 
+pub struct ArrayOfArrayOfNumberOnly(object);
```
Comment (bjgill) [inline samples/server/petstore/rust-server/src/models.rs]: Hm. Have been looking into the missing objects with a single property. I've added some logs that catch them as they pass through `postProcessModels` (printing out the datatype):

```
[main] WARN  o.o.c.languages.RustServerCodegen - Found a disappearing model: OuterNumber(OuterNumber), number
[main] WARN  o.o.c.languages.RustServerCodegen - Found a disappearing model: ArrayOfNumberOnly(ArrayOfNumberOnly), object
[main] WARN  o.o.c.languages.RustServerCodegen - Found a disappearing model: EnumClass(EnumClass), string
[main] WARN  o.o.c.languages.RustServerCodegen - Found a disappearing model: List(List), object
[main] WARN  o.o.c.languages.RustServerCodegen - Found a disappearing model: NumberOnly(NumberOnly), object
```

It looks as if those with datatype `object` turn up in `models.rs` with the datatype `object`, whilst those with other datatypes just disappear...

**Comment by bjgill** [comment]
OK - models with variables now seem to be working correctly. Models without properties, though, are still troublesome. It looks as if they're going missing between `postProcessModels` and `models.mustache`. I wonder if there is a particular field in their `CodegenModel`s that is set wrong? I'm going to continue investigating...

**Comment by bjgill** [comment]
Ah-ha! It looks like the single-property models have `isAlias` set to `true`. Presumably, something somewhere is filtering them out.

I'm not sure what the correct solution is, but I think I can get something working by setting `isAlias` to `false` and then tidying up the `dataType`.

**Comment by bjgill** [comment]
Currently not working:

- [ ] Responses with `schema: {type: string}`
- [ ] Enum strings in form data params
- [ ] Byte strings in form data parameters
- [ ] Binary string properties

However, those aside, we now produce code that compiles correctly. I think I'm now out of time to resolve the gaps for the moment, however. I'll tidy this up so that this could be merged as a barely acceptable initial effort.

**Comment by bjgill** [comment]
Right. I think this is now in an acceptable state to get merged in (assuming CI passes). This PR allows `rust-server` to produce valid Rust in openapi-generator. It's not comprehensive - https://github.com/OpenAPITools/openapi-generator/pull/290#issuecomment-396654016 tracks the known areas that I have not been able to re-add support for. 

I'm proposing to merge this PR in its current state, however, since it does work for most mainline cases, and so may be useful in itself, and also as a starting point for other people to work off. I'm also almost out of time to work on `rust-server` for the moment, and would like to leave things in a reasonable clean state.

Once merged, I'll create an issue detailing all the known outstanding work required.

Technical committee: @frol @farcaller

**Comment by wing328** [comment]
FYI. The CI failures have nothing to do with the change in this PR.

**Comment by wing328** [comment]
I've reviewed the change, which looks good to me.

I'll submit a PR later for enhancements, e.g. check for model/object in CodegenParameter, `isFile` check, etc

**Pull request merged**